src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs

```diff
@@ -130,6 +130,29 @@ internal TypeWithModifiers SubstituteType(TypeSymbol previous)
             return new TypeWithModifiers(result);
         }
 
+        /// <summary>
+        /// Same as <see cref="SubstituteType"/>, but with special behavior around tuples.
+        /// In particular, if substitution makes type tuple compatible, transform it into a tuple type.
```

Comment:
Yes, we do. See UnifyUnderlyingWithTuple_05 unit-test, for example.

---
In reply to: [62076202](https://github.com/dotnet/roslyn/pull/11047#discussion_r62076202) [](ancestors = 62076202)